spec/datadog/core/telemetry/logging_spec.rb

```diff
@@ -100,3 +109,49 @@
     end
   end
 end
+
+RSpec.describe Datadog::Core::Telemetry::Logging::DatadogStackTrace do
+  describe '.from' do
+    it do
+      exception = StandardError.new('Yo!')
+
+      result = described_class.from(exception)
+
+      expect(result).to be_nil
+    end
+
+    it do
+      exception = StandardError.new('Yo!')
+      exception.set_backtrace([])
+
+      result = described_class.from(exception)
+
+      expect(result).to be_nil
+    end
+
+    it 'returns redacted stack trace' do
+      gem_root = Datadog::Core::Telemetry::Logging::DatadogStackTrace::GEM_ROOT
+
+      exception = StandardError.new('Yo!')
+      exception.set_backtrace(
+        [
+          "#{gem_root}/lib/datadog/core/telemetry/logging.rb:1 in `report`",
+          '/foo/bar/baz.rb:1 in `baz`',
+          '/foo/bar.rb:1 in `bar`',
+          '/foo.rb:1 in `foo`',
+        ]
+      )
```

Comment:
Hmmm.... This test makes me a bit suspicious, because it just "assumes" the contents of `GEM_ROOT` are correct.

In particular, if I do this:

```ruby
          GEM_ROOT = Pathname.new("/potato").cleanpath.to_s
# or
          GEM_ROOT = Pathname.new("#{__dir__}/..").cleanpath.to_s
```

the test still passes.

The latter one is especially dangerous: it means that if someone decides to move all of telemetry to a different folder, and re-runs the tests, the test suite will pass but the `GEM_ROOT` will be incorrect (because it relies on the exact "depth" the file is in the folder structure).

Here's my suggestion: use the filtering on a real exception, and see if you get the right result:

```
    it 'returns redacted stack trace' do
      realistic_exception =
        begin
          Datadog::Core::Telemetry::Event::Base.new.type
          nil
        rescue NotImplementedError => e
          e
        end
      expect(realistic_exception).to_not be nil

      result = described_class.from(realistic_exception)

      expect(result).to start_with('/lib/datadog/core/telemetry/event.rb')
      expect(result).to end_with(',REDACTED')
    end
```

Although a bit weirder, this test will correctly fail in both of the situations I presented above.